accepted/0000-prometheus-tls.md

```diff
@@ -0,0 +1,238 @@
+- Feature Name: Prometheus TLS support
+- Start Date: 2021-02-11
+- RFC PR:
+
+# Summary
+
+[summary]: #summary
+
+Secure connection to Prometheus server and exporters.
+
+# Motivation
+
+[motivation]: #motivation
+
+## Prometheus
+
+In the default configuration it is presumed that untrusted users have access to
+the Prometheus HTTP endpoint. They have access to all time series information
+contained in the database, plus a variety of operational/debugging
+information [^1].
+
+Applications needing access to Prometheus API:
+
+* _Grafana_
+* _Alertmanager_
+
+Users needing access to Prometheus API/UI:
+
+* monitoring administrator
+
+## Exporters
+
+Exporters generally only talk to one configured client instance with a preset
+set of commands/requests, which cannot be expanded via their HTTP endpoint.
+Still, all exposed metrics are available to untrusted users.
+
+> Uyuni should support and automate setting up secure configuration of
+> Prometheus and exporters.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Prometheus
+
+Prometheus 2.24.0 has introduced TLS encryption as well as basic and
+client-certificate authentication [^2]. All available configuration options are
+described in official documentation [^3]. Basic authentication credentials are
+stored in the configuration file. Passwords are hashed with bcrypt.
+
+To follow current SUSE Manager security model [^4] we should consider
+implementing following default configuration:
+
+* Upgrade Prometheus to version >= 2.24.1.
+* Create and deploy CA signed Prometheus server certificate.
```

Comment:
Are we contemplating the case of users bringing their own CA? That is a possibility for general Uyuni.

Do we intend to offer generation as the only possibility, or as an option, or we plan to add "bring your own CA" at a later point?

Reply:
I've reworded that and marked as optional step.